### ISSUE TITLE
Strong name validation failed

### ISSUE THREAD

**Issue by luisgepeto** [rabbitmq/rabbitmq-dotnet-client#250]
Hello I have downloaded the latest RabbitMQ.Client library via nuget. I am trying to add my assembly to a database project to use it as a SQL CLR. However, when running my CLR Im getting the following error:

> A .NET Framework error occurred during execution of user-defined routine or aggregate "RabbitEnqueue": 
> System.IO.FileLoadException: Could not load file or assembly 'RabbitMQ.Client, Version=4.0.0.0, Culture=neutral, PublicKeyToken=89e7d7c5feba84ce' or one of its dependencies. Strong name validation failed. (Exception from HRESULT: 0x8013141A) ---> System.Security.SecurityException: Strong name validation failed. (Exception from HRESULT: 0x8013141A)
> System.Security.SecurityException: 
> 
> System.IO.FileLoadException: 
>    at UserDefinedFunctions.RabbitEnqueue(Int32 changeLogId)

I have tried running the verification of the strong name by running `sn -vf RabbitMQ.Client.dll` and get the following:

> Failed to verify assembly -- Strong name validation failed.

After this, I downloaded the dll directly form the site and this one is correctly strongly named. Why isn't the nuget dll not strongly named?

**Comment by kjnilsson** [comment]
This is surprising as the dlls are indeed strong named. I just verified a the net451 dll downloaded from nuget.org myself. Can you tell me the exact package version and dll you are using and how you are reference them in your project? As a side note I am not sure the client works well with SQL CLR. We certainly do no testing in that area at this time.

**Comment by michaelklishin** [comment]
Worth mentioning that if you'd like to use this client on SQL CLR, you can use the 3.6.5 release and not notice any difference. The biggest value proposition of 4.0.x is .NET Core support.

**Comment by michaelklishin** [comment]
@luisgepeto any updates on the question above?

**Comment by kjnilsson** [comment]
I suspect there isn't an actual issue here as we haven't had any other reports and I verified the strong naming in the NuGet package. I will close this for now. @luisgepeto please re-open with further information if you have it.

**Comment by nberglund** [comment]
Hey guys,
I just came across this via a post in the RabbiMQ Google group. The latest version of RabbitMQ.Client (5.0.1) works just fine with SQLCLR. Just deploy the client as well as your dll manually and do not rely on the  SQL Database project (that is devils tool!) :). Both dll's (client and yours) have to be deployed as UNSAFE. 

I have a blog post [here](http://www.nielsberglund.com/2017/02/11/rabbitmq-sql-server/), which shows how to do this. The post uses an earlier version of the client, but I just checked with the 5.0.1 version and it works just fine.
